### ISSUE TITLE
option to specify alternative location for `mktemp`

### ISSUE THREAD

**Issue by asbisen** [JuliaData/CSV.jl#988]
Just noticed that while reading gzipped files using `CSV.jl:read()` creates temporary files in `/tmp`. This causes issues in certain scenarios. [Ref](https://github.com/JuliaData/CSV.jl/blob/2c02394b4ce11ee673ed1b6d0c811a98316bc2f9/src/utils.jl#L308)

1. The default directory `/tmp` may not have sufficient capacity
2. The default directory `/tmp` might not be on a fast device (happens all the time on cloud VM)

I am using [FLoops.jl](https://github.com/JuliaFolds/FLoops.jl) to convert several CSV files and am encountering both of the issues. The workaround for me is simple, but would be nice to have this handled by CSV.jl by providing an argument for fast scratch space.

**Comment by quinnj** [comment]
Yeah, this definitely seems useful. Mind taking a stab at a PR? It would be fairly straightforward:
  * Add a new keyword argument allowed for `CSV.File`/`CSV.Rows`/`CSV.Chunks` and pass to `CSV.Context`
  * in context.jl, thread the new arg through and pass to `getsource` which lives in utils.jl
  * Thread the arg through to `buffer_to_tempfile` and pass like `mktemp(parent=tempdir)`
  * Add inline docs in keyworddocs.jl file (which gets used for CSV.File/CSV.Rows/CSV.Chunks
  * Add a simple test to show the keyword arg gets threaded through and used correctly
  * Add manual docs by updating `docs/reading.md` w/ new keyword arg
  * Add example using new keyword arg to `docs/example.md` and reference it in manual entry

Happy to help with any of the steps, but this would be a pretty straightforward feature to have a first go with!

**Comment by quinnj** [comment]
This was fixed in the 1.0 release by decompressing gzip in memory without temporary files ([#1196](https://github.com/JuliaData/CSV.jl/pull/1196)). [issue resolution reviewed by AI; please respond with objections or additional feedback/concerns/issues]
